aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -370,6 +371,44 @@ public class AwsProperties implements Serializable {
    */
   public static final String S3_WRITE_TAGS_PREFIX = "s3.write.tags.";
 
+  /**
+   * Used by {@link GlueCatalog} to tag objects when writing. To set, we can pass a catalog
+   * property.
+   *
+   * <p>For more details, see
+   * https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html
+   *
+   * <p>Example: s3.write.table-name-tag-enabled=true
+   */
+  public static final String S3_WRITE_TABLE_NAME_TAG_ENABLED = "s3.write.table-name-tag-enabled";
+
+  public static final boolean S3_WRITE_TABLE_NAME_TAG_ENABLED_DEFAULT = false;
+
+  /**
+   * Used by {@link GlueCatalog} to tag objects when writing. To set, we can pass a catalog
+   * property.
+   *
+   * <p>For more details, see
+   * https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html
+   *
+   * <p>Example: s3.write.namespace-name-tag-enabled=true
+   */
+  public static final String S3_WRITE_NAMESPACE_TAG_ENABLED = "s3.write.namespace-name-tag-enabled";
+
+  public static final boolean S3_WRITE_NAMESPACE_TAG_ENABLED_DEFAULT = false;
+
+  /**
+   * Used by {@link GlueCatalog}. The table name is used as a part of table specific catalog
```

Comment:
we should note this is the tag name that will be used by `{@link S3_WRITE_TAGS_PREFIX}`